src/monitor.py

```diff
@@ -0,0 +1,271 @@
+    if args.timestamp is None:
+        target_dir = None
+        for current_dir in working_dir.iterdir():
+            if target_dir is None:
+                target_dir = current_dir
+            elif target_dir.name < current_dir.name:
+                target_dir = current_dir
+    else:
+        target_dir = os.path.join(working_dir, args.timestamp)
+        if not os.path.exists(target_dir):
+            print(f"Error: Specified timestamp {args.timestamp} does not exist")
+            sys.exit(1)
+        target_dir = Path(target_dir)
+
+    status(target_dir, args.verbosity)
+
+
+def status(timestamp_dir: Path, verbosity: int):
+    """
+    Usage: observe the shared log for an exerciser test run and print status information
+    @param timestamp_dir: Path object to the root dir of an exerciser run
+    @param verbosity: int specifying how verbose the print stmts should be
+    """
+    shared_log = os.path.join(timestamp_dir, "shared_exerciser.log")
+    if not os.path.exists(shared_log):
+        print(f"Error: Shared log does not exist for test run {timestamp_dir}")
+        sys.exit(1)
+
+    # print time info for exerciser run being analyzed, and the current time
+    run_time = datetime.strptime(timestamp_dir.name, "%Y-%m-%d_%H-%M")
+    curr_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    print(f"Evaluating run from: {run_time}")
+    print(f"Current time is: {curr_time}")
+
+    # 2 dicts to store info on the status of the exerciser run
+    # 1 for expected tests (appears in working dir)
+    # 1 for unkown tests (doesn't appear in working dir, but does appear in shared log)
+    # both are dicts of dicts. each subdict stores info for a single test
+    # subdict fields are: submitted_resources, executed_resources,
+    # succeeded_resources, failed_resources, aborted_resources
+    expected_tests = {}
+    unknown_tests = {}
+    for item in timestamp_dir.iterdir():
+        # every dir represents a test
+        if item.is_dir():
+            test_dict = {
+                "submitted_resources": [],
+                "executed_resources": [],
+                "succeeded_resources": [],
+                "failed_resources": [],
+                "aborted_resources": [],
+            }
+            expected_tests[item.name] = test_dict
+
+    # clusters dict to store mapping of event cluster to test and associated procs
+    clusters = {}
+    event_log = JobEventLog(shared_log)
+    # loop through all events in shared event log, and filter for submit, execute, termination,
+    # and abortion events
+    for event in event_log.events(0):
+        # submit event: add test info to related dicts
+        if event.type is JobEventType.SUBMIT:
+            log_notes = event["LogNotes"]
+            colon_pos = log_notes.find(":")
+            if colon_pos != -1:
+                testname, resource = log_notes[colon_pos + 1 :].split(",")
```

Comment:
```suggestion
            if ":" in log_notes:
                testname, resource = log_notes.split(":")[1].split(",")
```
Is also suitable for this... will leave this up to you if you want to change